invenio_banners/models.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-Banners is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Models."""
+
+from datetime import datetime
+
+from flask import current_app
+from invenio_db import db
+from sqlalchemy_utils.models import Timestamp
+
+
+class Banner(db.Model, Timestamp):
+    """Defines a message to show to users."""
+
+    __tablename__ = "banners"
+    __versioned__ = {"versioning": False}
+
+    id = db.Column(db.Integer, primary_key=True)
+
+    message = db.Column(db.Text, nullable=False)
+    """The message content."""
+
+    url_path = db.Column(db.String(255), nullable=True)
+    """Define in which URL /path the message will be visible."""
+
+    category = db.Column(db.String(20), nullable=False)
+    """Category of the message, for styling messages per category."""
+
+    start_date = db.Column(db.DateTime, nullable=False, default=datetime.now())
```

Comment:
```suggestion
    start_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
```

- Nasty one here, by actually calling `datetime.now()` the default would be always the time of this file's evaluation
- The `Timestamp` mixin uses `datetime.utcnow`, and thus it might look confusing to see shifted timezones in the DB. This might need to be changed in other places too for consistency...

Reply:
Not sure I understood your second point. Normally (need to fix the bug), we always store UTC dates to avoid timezone issues...

Reply:
It's just that throughout the rest of the code `datetime.now()` is used to compare dates. I wasn't sure if it was on purpose, i.e. when setting a start/end date, one would be always thinking about "server time" (i.e. Geneva time), instead of converting mentally from UTC to local time.

I agree that we should be using UTC everywhere (same as the `Timestamp` mixin does for the `created`/`updated` fields). Not sure if the Flask-Admin view does so though when submitting its form.

Reply:
Let's discuss it whenever you have minute, I don't see an easy solution given that Flask-Admin is using `naive` DateTime.

<img width="1131" alt="pgAdmin_4" src="https://user-images.githubusercontent.com/2089455/96505421-62435780-1256-11eb-9034-2dde20720a72.png">